### PR TITLE
Formatting cleanup in bn_print.c

### PR DESCRIPTION
**Pull request by paulidale**
Predominately removal of unnecessary brackets.
One mask & cast fix too.

Depends on #4102 being merged first.

### CODE REVIEW

**Comment by paulidale** [comment]
This PR is only the second commit.

File: crypto/bn/bn_print.c [hunk 1]
```diff
@@ -306,16 +306,16 @@ int BN_print(BIO *bp, const BIGNUM *a)
...
-    if (BN_is_zero(a) && (BIO_write(bp, "0", 1) != 1))
+    if (BN_is_zero(a) && BIO_write(bp, "0", 1) != 1)
         goto end;
     for (i = a->top - 1; i >= 0; i--) {
         for (j = BN_BITS2 - 4; j >= 0; j -= 4) {
             /* strip leading zeros */
-            v = ((int)(a->d[i] >> (long)j)) & 0x0f;
-            if (z || (v != 0)) {
-                if (BIO_write(bp, &(Hex[v]), 1) != 1)
+            v = 0x0f & (int)(a->d[i] >> (long)j);
```
Comment (dot-asm) [inline crypto/bn/bn_print.c]: This one deserves mask-then-cast just as much as previous similar case. And no (long) of course...

**Comment by dot-asm** [comment]
This is going to need re-base after #4102 applied, right?

**Comment by paulidale** [comment]
Yes, it will require a rebase after #4102.

**Comment by paulidale** [comment]
Rebased.

File: crypto/bn/bn_print.c [hunk 2]
```diff
@@ -32,14 +32,14 @@ char *BN_bn2hex(const BIGNUM *a)
...
-        *(p++) = '-';
+        *p++ = '-';
     for (i = a->top - 1; i >= 0; i--) {
         for (j = BN_BITS2 - 8; j >= 0; j -= 8) {
             /* strip leading zeros */
-            v = ((int)(a->d[i] >> (long)j)) & 0xff;
-            if (z || (v != 0)) {
-                *(p++) = Hex[v >> 4];
-                *(p++) = Hex[v & 0x0f];
+            v = (int)(0xff & (a->d[i] >> j));
```
Comment (levitte) [inline crypto/bn/bn_print.c]: Why swap the `&` operands?  Considering the rest of OpenSSL uses the `var & 0xwhatever` form, this one looks odd...

File: crypto/bn/bn_print.c [hunk 3]
```diff
@@ -306,16 +306,16 @@ int BN_print(BIO *bp, const BIGNUM *a)
...
-    if (BN_is_zero(a) && (BIO_write(bp, "0", 1) != 1))
+    if (BN_is_zero(a) && BIO_write(bp, "0", 1) != 1)
         goto end;
     for (i = a->top - 1; i >= 0; i--) {
         for (j = BN_BITS2 - 4; j >= 0; j -= 4) {
             /* strip leading zeros */
-            v = ((int)(a->d[i] >> (long)j)) & 0x0f;
-            if (z || (v != 0)) {
-                if (BIO_write(bp, &(Hex[v]), 1) != 1)
+            v = (int)(0xff & (a->d[i] >> j));
```
Comment (levitte) [inline crypto/bn/bn_print.c]: Same here

File: crypto/bn/bn_print.c (hunk 2 above)
Comment (paulidale) [inline crypto/bn/bn_print.c]: Changed around.

**Comment by paulidale** [comment]
Thanks.

**Pull request closed without merging**